### PR TITLE
Feature/view model files

### PR DESCRIPTION
**Pull request by Lenkelheit**

### CODE REVIEW

File: Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs [hunk 1]
```diff
@@ -189,49 +211,174 @@ public System.Collections.Generic.IEnumerable<string> RedoActionNames
...
+        #endregion
 
-        
 
         // METHODS
         private void NewFileMethod(object o)
         {
-            throw new System.NotImplementedException();
+            canvas.Clear();
+            OnPropertyChanged(new PropertyChangedEventArgs("dataChanged"));
```
Comment (iamprovidence) [inline Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs]: there is no such property as "dataChanged".
there should be a name of property that is binding to view, as we bind all property there could be any of them, but not name of field.

File: Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs [hunk 2]
```diff
@@ -189,49 +211,174 @@ public System.Collections.Generic.IEnumerable<string> RedoActionNames
...
+        }
+        private void OpenFileMethod(object o)
+        {
+            try
+            {
+                if (dialogService.OpenFileDialog()) 
+                {
+                    currentFileName = dialogService.FilePath;
+                    fileService.Load(canvas, currentFileName);
+                    dialogService.ShowMessage("File is opened and data is loaded from it.");
```
Comment (iamprovidence) [inline Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs]: show message after action has been done is bad idea, too many messages would be shown.

File: Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs [hunk 3]
```diff
@@ -75,16 +86,17 @@ public ApplicationViewModel(IFileService fileService, IDialogService dialogServi
...
         // PROPERTIES
+        /// <summary>
+        /// Returns name of file.
+        /// </summary>
         public string FileName
         {
             get
             {
-                throw new System.NotImplementedException();
+                return currentFileName;
```
Comment (iamprovidence) [inline Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs]: you do not want to show all path to file, just file name

File: Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs (hunk 2 above)
Comment (iamprovidence) [reply Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs]: also do not forget OnPropertyChanged

File: Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs [hunk 4]
```diff
@@ -189,49 +211,174 @@ public System.Collections.Generic.IEnumerable<string> RedoActionNames
...
+            catch (System.Exception ex)
+            {
+                dialogService.ShowMessage(ex.Message);
+            }
+        }
+        private void SaveFileMethod(object o)
+        {
+            try
+            {
+                if (currentFileName.Contains("Pentagon Editor"))
```
Comment (iamprovidence) [inline Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs]: "contains" is bad idea, path can easily contain these words, if so you will have unexpected behaviour. much easiear to have variable currentFileName to be "null" from the start and chacking for that

File: Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs [hunk 5]
```diff
@@ -189,49 +211,174 @@ public System.Collections.Generic.IEnumerable<string> RedoActionNames
...
+            catch (System.Exception ex)
+            {
+                dialogService.ShowMessage(ex.Message);
+            }
+        }
+        private void ExitMethod(object o)
+        {
+            if (SavingBeforeClosingWindow() != System.Windows.MessageBoxResult.Cancel) 
+            {
+                ((System.Windows.Window)o).Close();
```
Comment (iamprovidence) [inline Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs]: just close application, there will not be any other windows

File: Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs [hunk 6]
```diff
@@ -189,49 +211,174 @@ public System.Collections.Generic.IEnumerable<string> RedoActionNames
...
+                if (boxResult == System.Windows.MessageBoxResult.Yes) 
+                {
+                    Save();
+                }
+                return boxResult;
+            }
+            return System.Windows.MessageBoxResult.None;
+        }
+        private void Save()
+        {
```
Comment (iamprovidence) [inline Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs]: nice try, but it should be implemented through command, its too complex, lets get rid of it

File: Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs (hunk 1 above)
Comment (Lenkelheit) [reply Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs]: Ok.

File: Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs (hunk 2 above)
Comment (Lenkelheit) [reply Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs]: Are all such messages unnecessary?

File: Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs (hunk 3 above)
Comment (Lenkelheit) [reply Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs]: Okey

File: Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs (hunk 6 above)
Comment (Lenkelheit) [reply Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs]: ok

File: Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs (hunk 4 above)
Comment (Lenkelheit) [reply Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs]: okay

File: Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs (hunk 2 above)
Comment (iamprovidence) [reply Task2/Project/DataControl/ViewModel/ApplicationViewModel.cs]: yep, they only are necessary if exception happens

**Pull request merged**